pkg/controller/pxc/controller.go

```diff
@@ -230,6 +230,61 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	if (o.Spec.HAProxy != nil && o.Spec.HAProxy.Enabled) &&
+		(o.Spec.ProxySQL != nil && o.Spec.ProxySQL.Enabled) {
+		return reconcile.Result{}, errors.New("Can't Enable Both HAProxy and ProxySQL please only select one of them")
+	}
```

Comment:
this check should be a part of https://github.com/percona/percona-xtradb-cluster-operator/blob/bc4f532d03508dd03165c2b2786b7a22cfd724a4/pkg/apis/pxc/v1/pxc_types.go#L325